src/vm-vcpu-ref/src/x86_64/interrupts.rs

```diff
@@ -43,23 +43,38 @@ pub const APIC_LVT1: usize = 0x360;
 pub const APIC_MODE_NMI: u32 = 0x4;
 pub const APIC_MODE_EXTINT: u32 = 0x7;
 
-pub fn get_klapic_reg(klapic: &kvm_lapic_state, reg_offset: usize) -> u32 {
+/// Errors associated with operations related to interrupts.
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// The register offset is invalid.
+    InvalidRegisterOffset,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+pub fn get_klapic_reg(klapic: &kvm_lapic_state, reg_offset: usize) -> Result<u32> {
     let range = reg_offset..reg_offset + 4;
-    let reg = klapic.regs.get(range).expect("get_klapic_reg range");
-    read_le_i32(&reg) as u32
+    let reg = klapic.regs.get(range).ok_or(Error::InvalidRegisterOffset)?;
+    Ok(read_le_i32(&reg) as u32)
 }
 
-pub fn set_klapic_reg(klapic: &mut kvm_lapic_state, reg_offset: usize, value: u32) {
+pub fn set_klapic_reg(klapic: &mut kvm_lapic_state, reg_offset: usize, value: u32) -> Result<()> {
+    // The value that we are setting is a u32, which needs 4 bytes of space.
+    // We're here creating a range that can fit the whole value.
     let range = reg_offset..reg_offset + 4;
-    let mut reg = klapic.regs.get_mut(range).expect("set_klapic_reg range");
-    write_le_i32(&mut reg, value as i32)
+    let mut reg = klapic
+        .regs
+        .get_mut(range)
+        .ok_or(Error::InvalidRegisterOffset)?;
+    write_le_i32(&mut reg, value as i32);
```

Comment:
Should we write a comment about why the conversion from u32 to i32 is safe? The same applies to `get_klapic_reg`.

Reply:
I just actually made everything i32 because that's what we end up using.